webcrawler.py

```diff
@@ -76,6 +77,21 @@ def collate():
                 'title': result_titles[url],
                 'keywords': str(', '.join(sorted(result_keywords[url]))),
             })
+            download(url)
```

Comment:
This is the wrong place to download, since at this point you haven't filtered or collated the URLs. You will also end up downloading the same URL multiple times (eg. if the same page shows up in different searches). A better place is just as you're about to write the output, when all the URLs are nicely collected.

Reply:
My bad - I misread where this code was. Ignore this comment.